src/webgpu/shader/validation/uniformity/uniformity.spec.ts

```diff
@@ -227,3 +227,129 @@ g.test('basics')
 
     t.expectCompileResult(t.params.expectation, code);
   });
+
+const kFragmentBuiltinValues = [
+  {
+    builtin: `position`,
+    type: `vec4<f32>`,
+  },
+  {
+    builtin: `front_facing`,
+    type: `bool`,
+  },
+  {
+    builtin: `sample_index`,
+    type: `u32`,
+  },
+  {
+    builtin: `sample_mask`,
+    type: `u32`,
+  },
+];
+
+g.test('fragment_builtin_values')
+  .desc(`Test uniformity of fragment built-in values`)
+  .params(u => u.combineWithParams(kFragmentBuiltinValues).beginSubcases())
+  .fn(t => {
+    let cond = ``;
+    switch (t.params.type) {
+      case `u32`:
+      case `i32`:
+      case `f32`: {
+        cond = `p > 0`;
+        break;
+      }
+      case `vec4<u32>`:
+      case `vec4<i32>`:
+      case `vec4<f32>`: {
+        cond = `p.x > 0`;
+        break;
+      }
+      case `bool`: {
+        cond = `p`;
+        break;
+      }
+      default: {
+        unreachable(`Unhandled type`);
+      }
+    }
+    const code = `
+@group(0) @binding(0) var s : sampler;
+@group(0) @binding(1) var tex : texture_2d<f32>;
+
+@fragment
+fn main(@builtin(${t.params.builtin}) p : ${t.params.type}) {
+  if ${cond} {
+    textureSample(tex, s, vec2<f32>(0,0));
```

Comment:
`textureSample` is marked `@must_use`, so this shader will be rejected even if the uniformity analysis isn't implemented correctly.

We try to avoid validation tests that unconditionally expect failures because of these sorts of issues - it's too easy to accidentally introduce unrelated validation errors that will go unnoticed. We generally try to have a control case that passes.

The control case here could be to use the diagnostic opt-out mechanism.